src/net.cpp

```diff
@@ -1608,6 +1608,43 @@ std::unordered_set<Network> CConnman::GetReachableEmptyNetworks() const
     return networks;
 }
 
+int CConnman::GetFullOutboundAndManualCount(Network net) const
+{
+    LOCK(m_nodes_mutex);
+    int outbound_peers{0};
+    std::optional<Network> net_alt;
+    // Treat clearnet (IPv4 and IPv6) as one network
+    if (net == Network::NET_IPV4) net_alt = Network::NET_IPV6;
+    if (net == Network::NET_IPV6) net_alt = Network::NET_IPV4;
+    for (const auto& pnode : m_nodes) {
+        if (pnode->fSuccessfullyConnected && !pnode->fDisconnect && (pnode->IsFullOutboundConn() || pnode->IsManualConn())) {
+            auto net_peer{pnode->addr.GetNetwork()};
+            if (net == net_peer || (net_alt.has_value() && net_alt == net_peer)) {
+                ++outbound_peers;
+            }
+        }
+    }
+    return outbound_peers;
+}
+
+bool CConnman::MaybePickPreferredNetwork(std::optional<Network>& network)
```

Comment:
The way this is implemented would give clearnet 2x chance of being returned over another network with 0 peers. Is this the intention?

For example: let all of IPv4, IPv6 and Tor be reachable and there are 0 peers to each of those networks. Clearnet (IPv4 or IPv6) will be returned in 67% of the cases and Tor in 33%. Should that not be clearnet 50%, Tor 50%?

Reply:
I think the requirement here is only "try each network with reasonable probability" -- you want to be connected to all of the networks, so which one comes first doesn't matter very much.

If you don't have ipv4 or ipv6 connections but are going through this codepath, then that means you have all your outbounds connected via privacy networks, but haven't chosen to `-onlynet=tor` or similar. So biassing towards clearnet probably makes sense anyway: clearnet connections probably have higher bandwidth/lower latency so will give you blocks/txs quicker, reducing the load on the privacy networks.

Reply:
Ok, I am just checking because elsewhere the intention is to treat clearnet as one network and this code does not do that. I do not have a strong opinion either way.